File: ambience/ambictlnet/ambictl/Ambience/LidlModule.cs
namespace Ambience
{
    public class LidlModule
    {
        private readonly string _absPath;
        public readonly string CmakeTarget;


        public LidlModule(string absPath, string cmakeTarget)
        {
            _absPath = absPath;
            CmakeTarget = cmakeTarget;
        }

        public string AbsoluteImport()
        {
            return _absPath;
        }

        public string FileName()
        {
            return System.IO.Path.GetFileName(_absPath);
        }

        public ServiceInterface GetService(string name)
        {
            return new ServiceInterface(this, name);
        }
    }
}

File: ambience/ambictl/ambictl/lidl_module.py
import ambictl.service_interface
import os


class LidlModule:
    _abs_path: str
    cmake_target: str

    def __init__(self, abs_path: str, cmake_target: str):
        self._abs_path = abs_path
        self.cmake_target = cmake_target

    def absolute_import(self):
        return self._abs_path

    def file_name(self):
        return os.path.basename(self._abs_path)

    def get_service(self, name: str) -> ambictl.ServiceInterface:
        return ambictl.ServiceInterface(self, name)
